tests/Control/SMIMEMailerTest.php

```diff
@@ -0,0 +1,243 @@
+
+class SMIMEMailerTest extends SapphireTest
+{
+
+    protected $usesDatabase = false; // phpcs:ignore
+
+    public function testEncryptionOnly()
+    {
+        $recipientEncryptionCrt = $this->getAsset('/assets/smime_test_recipient.crt');
+        $email = $this->buildEmail('Email with encryption', 'This is a confidential email.');
+        $partialMockMailer = $this->createPartialMockedMIMEMailer([$recipientEncryptionCrt]);
+
+        $partialMockMailer->expects($this->once())->method('sendSwift')->with(
+            self::callback(function ($message) {
+                $encryptedMessage = $message->toString();
+                self::assertStringContainsString('Content-Type: application/x-pkcs7-mime;', $encryptedMessage);
+                self::assertStringNotContainsString('This is a confidential email.', $encryptedMessage);
+
+                return true;
+            })
+        );
+
+        $partialMockMailer->send($email);
+    }
+
+    public function testSuccessfulDecryption()
+    {
+        $email = $this->buildEmail('Email with encryption', '<h1>This is a confidential email</h1>');
+        $partialMockMailer = $this->createPartialMockedMIMEMailer([
+            $this->getAsset('/assets/smime_test_recipient.crt')
+        ]);
+
+        $partialMockMailer->expects($this->once())->method('sendSwift')->with(
+            self::callback(function ($message) {
+
+                // Output email message to file
+                $file = @tempnam(__DIR__ . '/tmp', 'smimetest');
+                $bs = new Swift_ByteStream_FileByteStream($file, true);
+                $message->toByteStream($bs);
+
+                // Use openssl to extract private key, using password
+                $privateKey = openssl_pkey_get_private(
+                    file_get_contents($this->getAsset('/assets/smime_test_recipient.key')),
+                    'Test123!'
+                );
+
+                // Decrypt email message
+                $decryptedFile = @tempnam(__DIR__ . '/tmp', 'decrypted');
+                openssl_cms_decrypt(
+                    $file,
+                    $decryptedFile,
+                    file_get_contents($this->getAsset('/assets/smime_test_recipient.crt')),
+                    $privateKey
+                );
+
+                // Check that decrypted file contains expected string
+                self::assertStringContainsString(
+                    '<h1>This is a confidential email</h1>',
+                    file_get_contents($decryptedFile)
+                );
+
+                return true;
+            })
+        );
+
+        $partialMockMailer->send($email);
+    }
+
+    public function testSigningOnly()
+    {
+        $recipientSigningCrt = $this->getAsset('/assets/smime_test_sender.crt');
+        $recipientSigningKey = $this->getAsset('/assets/smime_test_sender.key');
+        $email = $this->buildEmail(
+            'Email smime signed by sender',
+            '<h1>This is not a confidential email.</h1>'
```

Comment:
Same as above, might be worth changing this to a string variable e.g. `$unencryptedContent`